lcls_tools/common/pyepics_tools/pyepicsUtils.py

```diff
@@ -1,4 +1,42 @@
+from datetime import datetime
+from time import sleep
+
+from epics import PV as epicsPV
+
+# These are the values that decide whether a PV is alarming (and if so, how)
 EPICS_NO_ALARM_VAL = 0
 EPICS_MINOR_VAL = 1
 EPICS_MAJOR_VAL = 2
 EPICS_INVALID_VAL = 3
+
+
+class PVInvalidError(Exception):
+    def __init__(self, message):
+        super(PVInvalidError, self).__init__(message)
+
+
+class PV(epicsPV):
+    def __init__(self, pvname):
+        super().__init__(pvname, connection_timeout=0.01)
+
+    def put(self, value, wait=False, timeout=30.0,
+            use_complete=False, callback=None, callback_data=None,
+            waitForPut=True):
+        super(PV, self).put(value, wait=True, timeout=timeout,
+                            use_complete=use_complete, callback=callback,
+                            callback_data=callback_data)
+
+        if waitForPut:
```

Comment:
What's going on down here? I'm too tired to understand.

Reply:
This waits for the PV to actually be set to the value you requested before returning (pyepics put doesn't check that the put went through so it breaks some sequential executions). Like, I want to wait for the RF to actually be on before I start increasing the amplitude